include/soci/odbc/soci-odbc.h

```diff
@@ -187,11 +187,58 @@ struct odbc_standard_use_type_backend : details::standard_use_type_backend,
 private:
     // Copy string data to buf_ and set size, sqlType and cType to the values
     // appropriate for strings.
+
+    // Build only for C++17 and later
+#if __cplusplus >= 201703L // C++17 or later
```

Comment:
I'm not sure what is the point of having a C++17 version if we also seem to have a fallback for the earlier language versions, what advantage does it have?

Note that if you really wanted to have a single function for both kinds of string, you could have it even in C++11, just using normal `if` and not `if constexpr`.

Reply:
Alright, I will throw it out.